spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/FindRefComparisonTest.java

```diff
@@ -0,0 +1,97 @@
+import edu.umd.cs.findbugs.SystemProperties;
+import edu.umd.cs.findbugs.annotations.Confidence;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcher;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
+
+public class FindRefComparisonTest extends AbstractIntegrationTest {
+
+    @AfterAll
+    static void tearDown() {
+        SystemProperties.removeProperty("findbugs.refcomp.reportAll");
+        setPriorityAdjustment(0); // reset priority
+    }
+
+    @Test
+    void testComparisonsNotEnabled() {
+        /**
+        * When there is no priority adjustment, and we enable ref comparison for
+        * all, we only get an integer ref comparison bug.
+        */
+        setPriorityAdjustment(0); // reset priority
+        SystemProperties.setProperty("findbugs.refcomp.reportAll", "true"); // enable ref comparison
+
+        performAnalysis("RefComparisons.class");
+        assertNumOfRefCompBugs(1);
+        assertRefCompBug("integerBadComparison", 4, Confidence.HIGH);
+    }
+
+    @Test
+    void testComparisonsWhenPropertyUnset() {
+        // When raising the priority, we get all ref comparison bugs.
+        setPriorityAdjustment(-1); // raise priority
+
+        SystemProperties.removeProperty("findbugs.refcomp.reportAll");
+        performAnalysis("RefComparisons.class");
+        assertNumOfRefCompBugs(3);
+        assertRefCompBug("integerBadComparison", 4, Confidence.HIGH);
+        assertRefCompBug("myClass1BadComparison", 11, Confidence.LOW);
+        assertRefCompBug("myClass2BadComparison", 18, Confidence.LOW);
+    }
+
+    @Test
+    void testComparisonsWhenReportAllIsNotEnabled() {
+        setPriorityAdjustment(-1); // raise priority
+        SystemProperties.setProperty("findbugs.refcomp.reportAll", "false");
+
+        performAnalysis("RefComparisons.class");
+        assertNumOfRefCompBugs(1);
+        assertRefCompBug("integerBadComparison", 4, Confidence.HIGH);
+    }
+
+    @Test
+    void testComparisonsWhenReportAllIsEnabled() {
+        setPriorityAdjustment(-1); // raise priority
+        SystemProperties.setProperty("findbugs.refcomp.reportAll", "true");
+
+        performAnalysis("RefComparisons.class");
+        assertNumOfRefCompBugs(3);
+        assertRefCompBug("integerBadComparison", 4, Confidence.HIGH);
+        assertRefCompBug("myClass1BadComparison", 11, Confidence.LOW);
+        assertRefCompBug("myClass2BadComparison", 18, Confidence.LOW);
+    }
+
+    private void assertNumOfRefCompBugs(int num) {
+        final BugInstanceMatcher bugTypeMatcher = new BugInstanceMatcherBuilder()
+                .bugType("RC_REF_COMPARISON").build();
+        assertThat(getBugCollection(), containsExactly(num, bugTypeMatcher));
+    }
+
+    private void assertRefCompBug(String method, int line, Confidence confidence) {
+        final BugInstanceMatcher bugInstanceMatcher = new BugInstanceMatcherBuilder()
+                .bugType("RC_REF_COMPARISON")
+                .inClass("RefComparisons")
```

Comment:
Are you sure that it's in this class every time? Even when it's in the inner class `RefComparisons$MyClass1`?

Reply:
hmm, I added a class parameter to this method to make this more explicitly, but I think it should be OK. I first check the total number of bugs (no matter the class) and then I use this method to check for each one of them.